src/FluentlyHttpClient/FluentHttpClient.cs

```diff
@@ -220,14 +232,62 @@ private HttpClient Configure(FluentHttpClientOptions options)
 
 		/// <inheritdoc />
 		public void Dispose()
-		{
-			RawHttpClient?.Dispose();
-		}
+			=> RawHttpClient?.Dispose();
+
+		public static implicit operator HttpClient(FluentHttpClient client)
+			=> client.RawHttpClient;
 
 		private static FluentHttpResponse ToFluentResponse(HttpResponseMessage response, IDictionary<object, object> items)
 			=> new FluentHttpResponse(response, items);
 
 		private static string DefaultSubClientIdentityFormatter(string parentId, string id)
 			=> $"{parentId}.{id}";
+
+		private class FluentlyHttpHandler : DelegatingHandler
+		{
+			private readonly IFluentHttpMiddlewareRunner _middlewareRunner;
+			private readonly Func<HttpResponseMessage, IDictionary<object, object>, FluentHttpResponse> _toFluentResponse;
+			private readonly FluentHttpClient _httpClient;
+			private readonly RequestTracker _requestTracker;
+
+			public FluentlyHttpHandler(
+				IFluentHttpMiddlewareRunner middlewareRunner,
+				Func<HttpResponseMessage, IDictionary<object, object>, FluentHttpResponse> toFluentResponse,
+				FluentHttpClient httpClient,
+				RequestTracker requestTracker,
+				HttpMessageHandler messageHandler = null
+			) : base(messageHandler ?? new HttpClientHandler())
+			{
+				_middlewareRunner = middlewareRunner;
+				_toFluentResponse = toFluentResponse;
+				_httpClient = httpClient;
+				_requestTracker = requestTracker;
+			}
+
+			protected override async Task<HttpResponseMessage> SendAsync(
+				HttpRequestMessage request,
+				CancellationToken cancellationToken
+			)
+			{
+				FluentlyExecutionContext context = null;
+				if (request.Properties.TryGetValue(RequestIdProperty, out var requestKey))
+				{
+					_requestTracker.TryPeek(requestKey.ToString(), out var fluentlyExecutionContext);
+					context = fluentlyExecutionContext;
+				}
+
+				var fluentlyRequest = context?.Request ?? request.ToFluentlyHttpRequest(_httpClient);
+
+				var response = await base.SendAsync(request, cancellationToken);
```

Comment:
this needs to go in the middleware runner - otherwise its sending before the middlewares, and if a middleware such as caching doesnt wont to trigger the send, it will be triggered anyway